internal/txapp/txapp.go

```diff
@@ -77,13 +77,16 @@ func (r *TxApp) GenesisInit(ctx context.Context, validators []*types.Validator,
 	}
 	defer tx.Rollback(ctx)
 
-	// Sanity Check: Verify that there is no data from previous instance
-	vals, err := getAllVoters(ctx, tx)
+	// Sanity Check: Check DB Height against Initial Height
+	// This is to ensure that the kwild is not started against stale database: at height 1
+	// If Blockchain is starting at height other than 1, db height should have state at initialHeight - 1
+	height, err := getDBHeight(ctx, tx)
+	fmt.Println("height: ", height, "error: ", err)
 	if err != nil {
-		return err
-	}
-	if len(vals) > 0 {
-		return errors.New("genesisInit: cannot initialize genesis state: database has data from previous reset instance")
+		// if the height is not found, we will assume that the database is empty
+		r.log.Info("GenesisInit: starting with empty database")
+	} else if initialHeight-1 != height {
+		return fmt.Errorf("genesisInit: expected database to be at height %d, got %d", initialHeight-1, height)
```

Comment:
Correct to error if `!=`, but if `==`, then we'd want this to be a no-op and return right away since we've done the init already.

However, I realized we're doing InitChain wrongish: https://github.com/cometbft/cometbft/issues/203